app/controllers/expenses_controller.rb

```diff
@@ -0,0 +1,38 @@
+class ExpensesController < ApplicationController
+  
+  def create
+    begin
+      expense = Expense.create(expense_params)
+      render status: :created, json: expense.to_json
+    rescue
+      render status: 400
+    end
+  end
+
+  def destroy
+    begin
+      params.require(:id)
+      Expense.destroy_by(id: params[:id])
+      render status: :ok
+    rescue
+      render status: 400
+    end
+  end
+
+  def update
+    begin
+      id = params[:id]  
+      expense = Expense.find_by(id: id)    
+      expense.update(expense_params)
+    rescue
```

Comment:
Let's not use an empty `rescue`. Then, use `find_by!` method to find the record and raise error if it is not found. Use another rescue to catch for `ActiveRecord::RecordNotFound` and return `404`.